utils/data_vis_utils.py

```diff
@@ -0,0 +1,432 @@
+            ],
+        )
+        .properties(width=600, height=400)
+    )
+    return chart
+
+
+def create_chart_average_annual_consumption(
+    profile_annual_avgs: pd.DataFrame, energy_type: str, colour: str = NESTA_COLOURS[0]
+) -> alt.Chart:
+    """
+    Create a bar plot showing the average annual energy consumption per profile.
+
+    Args:
+        profile_annual_avgs (pd.DataFrame): DataFrame containing average annual energy consumption per profile.
+        energy_type (str): Type of energy ('electricity' or 'gas').
+
+    Returns:
+        alt.Chart: Altair chart object showing the average annual energy consumption per profile.
+    """
+    data = profile_annual_avgs.copy()
+
+    y_col = f"avg_annual_{energy_type[:4]}_consumption_kWh"
+    title = f"Average annual {energy_type} consumption (kWh)"
+
+    if energy_type == "gas":
+        data = data[~data["profile"].isin(configs.profiles_low_gas_count)]
+
+    chart = (
+        alt.Chart(data, title=alt.TitleParams(text=title, anchor="middle"))
+        .mark_bar()
+        .encode(
+            x=alt.X("profile:O", title="Profile", axis=alt.Axis(labelAngle=0)),
+            y=alt.Y(f"{y_col}:Q", title=f"{energy_type} consumption (kWh)"),
+            color=alt.value(colour),
+            tooltip=[
+                alt.Tooltip("profile:O", title="Profile"),
+                alt.Tooltip(f"{y_col}:Q", title=title, format=".0f"),
+            ],
+        )
+        .properties(width=600, height=400)
+    )
+    return chart
+
+
+def plot_contextual_info(data: pd.DataFrame, variable: str, title: str) -> alt.Chart:
+    """
+    Creates a bar chart comparing the proportion of households with a specific
+    characteristic across different profiles.
+    It also adds a horizontal dashed line indicating the average value in the population.
+
+    Args:
+        data (pd.DataFrame): contextual data
+        variable (str): conrextual variable to plot (e.g. "income") withtout "counts_" or "proportion_" prefix
+        title (str): title of the chart
+
+    Returns:
+        alt.Chart: An Altair chart object showing the proportion of households with the specified characteristic
+    """
+    # In the contextual info df, the profile is stored as a string
+    profiles_as_strings = [str(p) for p in configs.profile_numbers]
+
+    # select only contextual information about profiles by select numbers only
+    # (when there are low counts for a number of profiles, contextual information
+    # is provided as an aggregate and "profile" might have values such as "All other profiles")
+    profiles_specific_data = data[data["profile"].isin(profiles_as_strings)]
+
+    # creating barchart with a bar for each profile
+    chart = (
+        alt.Chart(
+            profiles_specific_data, title=alt.TitleParams(text=title, anchor="middle")
+        )
+        .mark_bar()
+        .encode(
+            x=alt.X("profile:N", axis=alt.Axis(labelAngle=0)),
```

Comment:
```suggestion
            x=alt.X("profile:N", axis=alt.Axis(labelAngle=0), sort=[str(i) for i in range(1, 11)]),
```

Suggestion to ensure profiles 1-10 are shown in the right order (related to Roisin's comment).